src/particle_data.cpp

```diff
@@ -12,6 +15,21 @@
 
 namespace openmc {
 
+void Geometron::mark_as_lost(const std::string& message)
+{
+  mark_as_lost(message.c_str());
+}
+
+void Geometron::mark_as_lost(const std::stringstream& message)
+{
+  mark_as_lost(message.str());
+}
+
+void Geometron::mark_as_lost(const char* message)
+{
+  fatal_error(message);
+}
```

Comment:
At first glance, this seems problematic as the function is very different than the `Particle::mark_as_lost()` version, but the Geometron version is called even when tracking MC particles in some locations. Thus, it seems like it may be the case that certain types of lost particle events that previously just resulted in a warning message (not that uncommon for super large runs) would now result in a fatal error?